phoenix-core/src/main/java/org/apache/calcite/jdbc/PhoenixCalciteFactory.java

```diff
@@ -161,6 +162,29 @@ public Object apply(
             }));
         }
 
+        public CalciteStatement createStatement(String sql, int resultSetType, int resultSetConcurrency, int resultSetHoldability) throws SQLException {
+            try {
+                return super.createStatement(resultSetType, resultSetConcurrency, resultSetHoldability);
+            } catch (SQLException e) {
+                if (e.getCause().getCause() instanceof SQLException) {
+                    throw (SQLException) e.getCause().getCause();
```

Comment:
Can we can have a utility method that unwraps the Exception? And as we might different number of levels of Exception wrapping, can we just look down one level at a time till we hit an instance of SQLException?